ebmlite/core.py

```diff
@@ -1412,45 +1426,140 @@ def _crawl(el):
 #
 # ==============================================================================
 
+def _expandSchemaPath(path, name=''):
+    """ Helper function to process a schema path or name, converting module
+        references to Paths.
+
+        @param path: The schema path. May be a directory name, a module
+            name in braces (e.g., `{idelib.schemata}`), or a module
+            instance. Directory and module names may contain schema
+            filenames.
+        @param name: An optional schema base filename. Will get appended
+            to the resulting `Path`/`Traversable`.
+        @return: A `Path`/`Traversable` object.
+    """
+    strpath = str(path)
+    subdir = ''
+
+    if not strpath:
+        path = strpath = os.getcwd()
+    elif '{' in strpath:
+        if '}' not in strpath:
+            raise IOError(errno.ENOENT, 'Malformed module path', strpath)
+
+        m = re.match(r'(\{.+\})[/\\](.+)', strpath)
+        if m:
+            path, subdir = m.groups()
+            strpath = path
```

Comment:
Since curly braces are allowed by windows paths, I think we could end up in a situation where this raises a bad error.  We could probably lean more heavily into the regex with something like `r'^\{([^{}]+)}([/\\].[^/\\]+)$'` ahead of the if block and use that as a condition to test.  We could also probably validate the syntax of the module name at the same time, but that might also not be necessary.

Reply:
How much of a risk do filenames containing braces pose? There are two cases: literal paths being mistaken for package references, and package references being mistaken for literal paths. In the first case, the paths containing brackets would have to be in the PYTHONPATH and contain a `__init__.py` (or end in `.py` if it's a file); the only time I've seen braces in names are some temporary Windows system subfolders, and those aren't valid aren't valid Python package names. In the second case, the user just wouldn't be able to import out of a relative directory with brackets in the name, which seems like even more of an edge case.

The second case seems slightly more probable. It could be handled by adding the ability to escape brackets; maybe double brackets (since \ is the Windows path delimiter)?

Incidentally, my last commit adds the ability to reference subdirectories in package-relative paths, e.g. `"{idelib}/schemata/mide_ide.xml"`, which the example regex won't match. I'm honestly not sure if it should be allowed, though I think the behavior may be what users would expect.